sparse/sfold.go

```diff
@@ -33,6 +37,22 @@ func FoldFile(childFileName, parentFileName string) error {
 		panic("file sizes are not equal")
 	}
 
+	go coalesce(parentFileName, childFileName, ops)
+
+	return nil
+}
+
+func coalesce(parentFileName, childFileName string, ops FoldFileOperations) (err error) {
+	var progress int
+
+	defer func() {
+		if err != nil {
+			ops.UpdateFoldFileProgress(progress, true, err)
```

Comment:
Since previous we have logs for each statement, it's better we keep it that way to print log when an error happens as well.